source/coding-the-legislation/legislation_parameters.md

```diff
@@ -95,43 +95,55 @@ public_sector:
+#### Scales
 
-Scales are constituted of brackets. Brackets are defined by amounts, bases, rates, average rates and thresholds.
+Scales are complex parameters constituted of brackets. They offer convenient built-in tools for recurring calculation patterns.
 
-Sample `parameters/tax_on_salary/tax_scale.yaml`:
+For instance, a marginal rate test scale can be defined in a YAML file with:
+
+File `parameters/tax_on_salary/tax_scale.yaml`
 ```yaml
 description: Scale for tax on salaries
 brackets:
-- rate:
-    1950-01-01:
-      value: 0.0
-    2010-01-01:
-      value: 0.02
-  threshold:
-    1950-01-01:
-      value: 0.0
-- rate:
-    1950-01-01:
-      value: 0.2
-  threshold:
-    1950-01-01:
-      value: 2000
+  - rate: # Define a 1st bracket
+      1950-01-01:
+        value: 0.0  # The rate applied to the first bracket was 0% from 1950 to 2009
+      2010-01-01:
+        value: 0.02  # The rate applied to the first bracket has changed to 2% in 2010
+    threshold:
+      1950-01-01:
+        value: 0.0
+  - rate: # Define a 2nd bracket
+      1950-01-01:
+        value: 0.2  # The rate applied to the second bracket has been 20% since 1950
+    threshold:
+      1950-01-01:
+        value: 2000  # The 2nd bracket starts for wages beyond 2000
+metadata:
+  threshold_unit: currency-EUR
+  rate_unit: /1
 ```
 
-Example: [the french tax scale on income](https://fr.openfisca.org/legislation/impot_revenu.bareme)
-
-#### Import parameters from IPP tables
-
-> This section applies only to OpenFisca-France.
+It can then be used in a formula with:
+```py
+def formula(person, period, parameters):
+    salary = person('salary', period)
+    scale = parameters(period).tax_on_salary.tax_scale
+    return scale.calc(salary)
+```
 
-The [<abbr title="Institut des politiques publiques">IPP</abbr>](http://www.ipp.eu/) is a French centre in economics which produces tax and benefit tables in the `XLSX` format, with parameters history.
+Other built-in scales are:
+- Marginal amount tax scale:
+  - Matches the input amount to a set of brackets and returns the sum of cell values from the lowest bracket to the one containing the input
+  - Defined as in the previous YAML example, but replacing `rate` by `amount`
+- Single amount tax scale:
+  - Matches the input amount to a set of brackets and returns the single cell value that fits within that bracket
+  - Defined as in the previous YAML example, but replacing `rate` by `amount`, and adding a `type: single_amount` to the parameter's metadata
 
-The OpenFisca team works on importing those data into the YAML parameter files of OpenFisca-France.
 
-See [this README](https://github.com/openfisca/openfisca-france/tree/master/openfisca_france/scripts/parameters/baremes_ipp) for more information.
+Example: [the french tax scale on income](https://fr.openfisca.org/legislation/impot_revenu.bareme)
 
-## Computing a parameter that depends on a variable (fancy indexing)
+#### Parameter depending on a variable (fancy indexing)
```

Comment:
> I thinks this qualifies as an "advanced usecase"

Reply:
I like the previous version of this section title better, because it starts with a gerund, which is a nice pattern. "Parameter depending" is much less clear gramatically.

Reply:
Ok with keeping gerunds.